City-of-Helsinki/paatokset-search
# PP-363: Decisions search filter fixes

**Drupal setup**
- On your local drupal installation, checkout this branch: `PP-363-search-filter-fixes`
- Run `make drush-cim drush-cr`
- If you don't have any data, run `make ahjo-migrations` (NOTE: you'll need the local proxy URL for this one. Also, edit the user permissions on the test site and add the following permissions for anonymous users: "Access external documents from Ahjo API")
- Rebuild and reindex the decisions index: https://helsinki-paatokset.docker.so/fi/admin/config/search/search-api/index/decisions

**To test**
- Checkout branch and start the app with `npm start` (make sure the index is set for decisions)
- Read the code changes. This has a lot of changes to the category and DM selection, so those parts of the code should especially be checked

**Date as selected filter**
- Test the date selector. The selected values should appear as a tag in the selected filters area
- You should be able to reset the date search by deselecting the tag

**All decisionmakers filter, deselecting decisionmakers**
- There should be a new item in the decisionmakers list: "Kaikki päättäjät"
  - Selecting this acts the same as having the filter empty, but should appear as a tag and shouldnt cause any errors
- You should now be able to deselect the selected item in the decisionmaker dropdown by clicking it again

**ID based decisionmaker and category selection**
- Decisionmaker and top category selection should now be based on the indexed "Sector ID" and "Top category code" fields
- Compare the fi and sv search results to the version on the test site:
  - https://nginx-paatokset-test.agw.arodevtest.hel.fi/fi/paattajat
  - https://nginx-paatokset-test.agw.arodevtest.hel.fi/sv/beslutsfattare
- On the test site, you should get vastly different results when filtering by category and decision maker (not the special case dm selections)
- On the local search app, the results should be more similar between language versions (they might vary by a few decisions, though)
- Also test that the URL query parameters work by reloading the page after adding filters
